otter/test/utils.py

```diff
@@ -845,3 +846,39 @@ def as_effect(self):
 def noop(_):
     """Ignore input and return None."""
     pass
+
+
+def intent_func(fname):
+    """
+    Return function that returns Effect of tuple of fname and its args. Useful
+    in writing tests that expect intent based on args
+    """
+    return lambda *a: Effect((fname,) + a)
+
+
+class Cache(object):
+    """ IScalingGroupServersCache impl for testing """
+
+    def __init__(self, tid, gid):
+        self.tid = tid
+        self.gid = gid
+
+    def ids(self, s):
+        return "cache" + s + self.tid + self.gid
```

Comment:
Non-blocking - might be easier to read and understand what this is in the tests if this were something like `"cache({},{}).{}".format(self.tid, self.gid, s)`, and `s` were one of (get, insert, delet).

So that the tests would use something like `"cache(tenant,group).get"` in the sequence dispatchers.  This would make the string longer though, and would result in more wrapping.

We can file a tech debt ticket, or just ignore it.  :)